src/tests/functional/navigation_tests.js

```diff
@@ -307,6 +307,18 @@ test("clicking the forward button", async ({ page }) => {
   assert.equal(await visitAction(page), "restore")
 })
 
+test("form submissions that redirect to a different location have a default advance action", async ({ page }) => {
```

Comment:
I would add a page refresh test to validate that morphing doesn't trigger if you click on a link pointing to the current page.